### ISSUE TITLE
unexpected dml statement found in query event

### ISSUE THREAD

**Issue by qiufengsese** [pingcap/tidb#7706]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
上游MySQL端binlog_format已经改为ROM，使用Syncer进行同步，然后报错。报错的表没在Syncer的配置文件中。报错信息是：2018/09/14 18:44:19 syncer.go:695: [error] fail to be parsed, error query UPDATE QRTZ_SCHEDULER_STATE SET LAST_CHECKIN_TIME = 1536919749922 WHERE SCHED_NAME = 'xxx' AND INSTANCE_NAME = 'yyy': unexpected dml statement found in query event。
配置文件内容为：
[[replicate-do-table]]
db-name ="db1"
tbl-name = "table1"

[[replicate-do-table]]
db-name ="db1"
tbl-name = "table2"

2. What did you expect to see?
使用Syncer正常同步数据

3. What did you see instead?

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
Release Version: v2.0.7
Git Commit Hash: 29ec059cb3b7d14b6f52c2f219f94a89570162bc
Git Branch: release-2.0
UTC Build Time: 2018-09-07 12:36:02
GoVersion: go version go1.11 linux/amd64
TiKV Min Version: 2.0.0-rc.4.1

谢谢！

**Comment by ciscoxll** [comment]
@qiufengsese  Thank you for your report.

**Comment by jackysp** [comment]
Thanks for your report, @qiufengsese ! PTAL @holys @july2993

**Comment by csuzhangxc** [comment]
@qiufengsese 
this error indicate that DML statement found in QUERY EVENT (this should not happen in ROW  binlog_format).

you changed binlog_format to ROW, but your syncer's startup pos specified in meta file may be earlier than the pos when you do the change.

change binlog_format only affect upcoming events, previous saved binlog events will not be converted.
so, you should set a new startup pos after the changing you did.

`replicate-do-table` rules applied after binlog events have been parsed in Syncer.
in other words, event table not need to sync, we still need to parse the binlog event.

**Comment by zhexuany** [comment]
@qiufengsese friendly ping. Please close this issue if the problem has been resolved.

**Comment by qiufengsese** [comment]
已定位到问题原因：改变binlog_format之后没有重置所有连接，导致有些库还是以statement的格式写binlog。十分感谢你们及时的解答。攒！

**Comment by chenmin1992** [comment]
how to skip these parse errors??? @csuzhangxc @jackysp
